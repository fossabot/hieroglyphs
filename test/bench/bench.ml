open Core_bench.Bench
module Command = Core.Command

let suite = Hiero.suite @ Rsa_pss.suite @ Secp.suite

let _ = Command.run (make_command suite)
