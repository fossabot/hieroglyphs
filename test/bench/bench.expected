Estimated testing time 2m (12 benchmarks x 10s). Change using -quota SECS.
┌─────────────────────────────────────────────────────┬─────────────────┬───────────────┬─────────────┬────────────┬────────────┐
│ Name                                                │        Time/Run │       mWd/Run │    mjWd/Run │   Prom/Run │ Percentage │
├─────────────────────────────────────────────────────┼─────────────────┼───────────────┼─────────────┼────────────┼────────────┤
│ hieroglyphs + blake2b ------ private key generation │ 36_115_048.25ns │ 1_257_186.16w │  24_212.48w │ 24_212.48w │    100.00% │
│ hieroglyphs + blake2b ------ message signing        │ 11_213_663.99ns │   552_974.77w │ 160_025.25w │ 28_670.25w │     31.05% │
│ hieroglyphs + blake2b ------ public key derivation  │  3_008_022.90ns │   190_464.86w │   7_971.99w │  7_971.99w │      8.33% │
│ hieroglyphs + blake2b ------ signature verification │ 12_598_340.69ns │   865_711.60w │  44_975.21w │ 44_975.21w │     34.88% │
│ rsa pss/sha256 1024 bits --- private key generation │ 35_080_509.59ns │       870.80w │       0.35w │      0.35w │     97.14% │
│ rsa pss/sha256 1024 bits --- public key derivation  │         11.46ns │         3.00w │             │            │            │
│ rsa pss/sha256 1024 bits --- message signing        │    771_613.48ns │     1_713.42w │       0.63w │      0.63w │      2.14% │
│ rsa pss/sha256 1024 bits --- signature verification │    142_191.97ns │     1_547.02w │       0.36w │      0.36w │      0.39% │
│ secp256k1 + sha256 hash ---- private key generation │     12_920.47ns │       355.71w │             │            │      0.04% │
│ secp256k1 + sha256 hash ---- public key derivation  │    214_130.84ns │        13.00w │             │            │      0.59% │
│ secp256k1 + sha256 hash ---- message signing        │    302_456.71ns │       230.00w │             │            │      0.84% │
│ secp256k1 + sha256 hash ---- signature verification │    361_341.90ns │       214.00w │             │            │      1.00% │
└─────────────────────────────────────────────────────┴─────────────────┴───────────────┴─────────────┴────────────┴────────────┘
